File: java/dubbo-demo/dubbo-demo-api/src/main/java/com/alibaba/dubbo/demo/UserResponse.java
package com.alibaba.dubbo.demo;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

/**
 * Created by master on 21/11/2017.
 */
public class UserResponse implements Serializable {
    private String status;
    private Map<String, String> info;
    private GenerticTypeResponse<Long> list;
    private HashMap<String, String> some;

    public String getStatus() {
        return status;
    }

    public HashMap<String, String> getSome() {
        return some;
    }

    public void setSome(HashMap<String, String> some) {
        this.some = some;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Map<String, String> getInfo() {
        return info;
    }

    public void setInfo(Map<String, String> info) {
        this.info = info;
    }

    public GenerticTypeResponse<Long> getList() {
        return list;
    }

    public void setList(GenerticTypeResponse<Long> list) {
        this.list = list;
    }

    @Override
    public String toString() {
        return "UserResponse{" +
                "status='" + status + '\'' +
                ", info=" + info +
                ", list=" + list +
                '}';
    }
}

File: packages/dubbo/src/__tests__/providers/com/alibaba/dubbo/demo/UserResponse.ts
import {GenerticTypeResponse} from './GenerticTypeResponse';
import java from 'js-to-java';

//generate by interpret-cli dubbo2.js

export interface IUserResponse {
  some?: {[name: string]: string};
  list?: GenerticTypeResponse<number>;
  status?: string;
  info?: {[name: string]: string};
}

export class UserResponse {
  some?: {[name: string]: string};
  list?: GenerticTypeResponse<number>;
  status?: string;
  info?: {[name: string]: string};

  constructor(params: IUserResponse) {
    this.some = params.some;
    this.list = params.list;
    this.status = params.status;
    this.info = params.info;
  }

  __fields2java() {
    let someMapTransfer = new Map();
    for (let mapKey in this.some) {
      someMapTransfer.set(java.String(mapKey), java.String(this.some[mapKey]));
    }
    let infoMapTransfer = new Map();
    for (let mapKey in this.info) {
      infoMapTransfer.set(java.String(mapKey), java.String(this.info[mapKey]));
    }
    return {
      $class: 'com.alibaba.dubbo.demo.UserResponse',
      $: {
        some: java.HashMap(someMapTransfer),
        list: this.list ? this.list.__fields2java() : null,
        status: java.String(this.status),
        info: java.Map(infoMapTransfer),
      },
    };
  }
}
